src/test/a11y/customChecks.ts

```diff
@@ -0,0 +1,190 @@
+    for (let n = 0; n < tasks.length; n++) {
+      // expect each anchor(task) has it corresponding status
+      expect(document.getElementById(`${tasks[n].getAttribute('aria-describedby')}`), `Task "${tasks[n].textContent.trim()}" does not contain corresponding status`).to.not.equal(null)
+    }
+  }
+}
+
+// validate checkAnswers page as per the recommended structure @ https://design-system.service.gov.uk/patterns/check-answers/
+export const checkAnswers = (window: Window, document: Document) => {
+  // this expects and validates a html structrue like below
+  /**
+   * <dl>
+   *  <div>
+   *    <dt>Full name</dt>
+   *    <dd>Jonatha</dd>
+   *    <dd> <a href="/path/to/edit/full-name">Change <span class="visually-hidden">full name</span></a></dd>
+   *  </div>
+   * </dl>
+   */
+  const definitionList = document.getElementsByTagName('dl')
+  expect(definitionList.length, 'Task-list page should contain atleast one <dl> tag with list of answers provided').to.be.greaterThan(0)
+
+  for (let i = 0; i < definitionList.length; i++) {
+    const childElements = definitionList[i].children
+    for (let c = 0; c < childElements.length; c++) {
+      // compare first child text with last child's link text
+      const dt = childElements[c].children[0].textContent.trim().toLocaleLowerCase() // content of <dt>...</dt>
+      const link = childElements[c].children[2].querySelector('a') // <a href="/path...">...</a> or null
+      if (link) {
+        expect('Change ' + dt).to.equal(link.textContent.trim())
+      } else {
+        console.log('INFO: no change link available for "' + dt + '"')
+      }
+    }
+  }
+}
+
+/**
+ *
+ * @param window
+ * @param document
+ * expect error container with class 'error-summary' is present
+ * expect h2 tag for error messages
+ * expect 'There was a problem' text in h2
+ * only one h1 tag per page
+ */
+export const checkError = (window: Window, document: Document) => {
+  const className = 'error-summary'
+  const errorContainer = document.getElementsByClassName(className)
+  // should contain only one error-summary
+  expect(errorContainer.length, `div element with '${className}' must be present`).to.be.equal(1)
+  // error-summary div should contain <h2>There was a problem</h2>
+  const heading = errorContainer[0].getElementsByTagName('h2')
+  expect(heading.length, 'Error heading should use h2 tag').to.be.eq(1)
+  expect(heading[0].textContent.trim()).to.be.eq('There was a problem')
+}
+
+export const checkRole = (window: Window, document: Document) => {
+  // this expects and validates 'Total monthly income' 'p' tag has role="status" and aria-live="polite"
+  /**
+   * <p class="heading-small calculation-outcome-container" role="status" aria-live="polite">
+   *  {{ t('Total monthly income:') }} £<span class="total-monthly-income-expense"> {{ totalMonthlyIncomeExpense | default('0.00') }} </span>
+   * </p>
+   */
+  const definitionList = document.getElementsByClassName('calculation-outcome-container')
```

Comment:
Can we change const name from 'definitionList' to something more appropriate to the test like 'messageElement' or 'elementWithTotal'

Reply:
done updated as **elementWithTotal**